pre_commit_hooks/leading_tabs_remover.py

```diff
@@ -0,0 +1,48 @@
+import argparse
+import re
+
+MAX_REPLACE = 10
+TAB_WIDTH = 4
+
+
+def fix_file(filename, tab_width=TAB_WIDTH, max_replace=MAX_REPLACE):
+    with open(filename, 'r') as fd:
+        original_lines = fd.readlines()
+    new_lines = []
+    changed = False
+    regex = re.compile(r'^\s*\t')
+    for line in original_lines:
+        replace_count = 0
+        while regex.match(line) and replace_count < max_replace:
+            changed = True
+            replace_count += 1
+            line = line.replace('\t', ' ' * tab_width, 1)
+        if not replace_count < max_replace:
+            line = line.strip('\r\n')
+            raise RuntimeError('Reached max tab replacements for one line '
+                               f'({max_replace}) in file {filename}, on the '
+                               f'following line: "{line}". '
+                               'Aborting to avoid infinite loop.')
+        new_lines.append(line)
+    if changed:
+        print(f'Fixing {filename}')
+        with open(filename, 'w') as fd:
+            fd.write(''.join(new_lines))
+
+
+def main(args=None):
+    if args is None:
+        parser = argparse.ArgumentParser()
+        parser.add_argument('filenames', nargs='*')
+        args = parser.parse_args()
+    try:
+        for filename in args.filenames:
+            fix_file(filename)
```

Comment:
Allow use of max_replace and tab_width arguments?
```suggestion
        parser.add_argument('--tab_width', type=int, default=4)
        parser.add_argument('--max_replace', type=int, default=10)  
        args = parser.parse_args()
    try:
        for filename in args.filenames:
            fix_file(filename, tab_width=args.tab_width,
                     max_replace=args.max_replace)
```

Reply:
Sure, I'll do this, a little differently because we're refactoring to drop the need for the max_replace parameter. I didn't prioritize it because I don't know if/how argument options are used in `pre-commit`

Reply:
Do you know how command-line args work with pre-commit, by the way, @ZryletTC?

Reply:
Yeah, it gives a lot of options to pass in args. For us, we could just use the `args` property in the config yaml.
https://pre-commit.com/#passing-arguments-to-hooks